RecoPPS/Local/src/CTPPSDiamondRecHitProducerAlgorithm.cc

```diff
@@ -39,9 +39,9 @@ void CTPPSDiamondRecHitProducerAlgorithm::build(const CTPPSGeometry& geom,
     z_pos = det->parentZPosition();  // retrieve the plane position;
 
     // parameters stand for half the size
-    const float x_width = 2.0 * det->params().at(0);
-    const float y_width = 2.0 * det->params().at(1);
-    const float z_width = 2.0 * det->params().at(2);
+    const float x_width = 2.0 * det->getDiamondDimensions().xHalfWidth;
```

Comment:
There are unneeded repeated creations of the DiamondDimension object here, you can do:

```C++
const auto& diamondDim = det->getDiamondDimensions();
const float x_width = 2.0 * diamondDim.xHalfWidth;
const float y_width = 2.0 * diamondDim.yHalfWidth;
const float z_width = 2.0 * diamondDim.zHalfWidth;

```